lib/spack/spack/test/stage.py

```diff
@@ -699,3 +699,37 @@ def test_stage_constructor_with_path(self, tmpdir):
         testpath = str(tmpdir)
         with Stage('file:///does-not-exist', path=testpath) as stage:
             assert stage.path == testpath
+
+    def test_diystage_path_none(self):
+        """Ensure DIYStage for path=None behaves as expected."""
+        with pytest.raises(ValueError):
+            DIYStage(None)
+
+    def test_diystage_path_invalid(self):
+        """Ensure DIYStage for an invalid path behaves as expected."""
+        with pytest.raises(spack.stage.StagePathError):
+            DIYStage('/path/does/not/exist')
+
+    def test_diystage_path_valid(self, tmpdir):
+        """Ensure DIYStage for a valid path behaves as expected."""
+        path = str(tmpdir)
+        stage = DIYStage(path)
```

Comment:
FWIW, another thing you could test is placing a file in this stage, and ensure that creating a stage in the directory with an existing file doesn't remove it (the point is that generally `Stage` objects are allowed to modify the contents of their associated root, but `DIYStage` is supposed to leave it alone).